crates/resolver/src/hosts.rs

```diff
@@ -47,14 +49,62 @@ impl Hosts {
     /// Look up the addresses for the given host from the system hosts file.
     pub fn lookup_static_host(&self, query: &Query) -> Option<Lookup> {
         if !self.by_name.is_empty() {
-            if let Some(val) = self.by_name.get(query.name()) {
-                let result = match query.query_type() {
-                    RecordType::A => val.a.clone(),
-                    RecordType::AAAA => val.aaaa.clone(),
-                    _ => None,
-                };
-
-                return result;
+            match query.query_type() {
+                RecordType::A | RecordType::AAAA => {
+                    if let Some(val) = self.by_name.get(query.name()) {
+                        let result = match query.query_type() {
+                            RecordType::A => val.a.clone(),
+                            RecordType::AAAA => val.aaaa.clone(),
+                            _ => None,
+                        };
+
+                        return result;
+                    }
+                }
+                RecordType::PTR => {
+                    if let Ok(ip) = query.name().parse_arpa_name() {
+                        let ip = ip.addr();
+                        let mut hostnames = self
+                            .by_name
+                            .iter()
+                            .filter(|(_, v)| match ip {
+                                IpAddr::V4(ip) => match v.a.as_ref() {
+                                    Some(lookup) => lookup.iter().any(|r| {
+                                        r.ip_addr().map(|it| it == ip).unwrap_or_default()
+                                    }),
+                                    None => false,
+                                },
+                                IpAddr::V6(ip) => match v.aaaa.as_ref() {
+                                    Some(lookup) => lookup.iter().any(|r| {
+                                        r.ip_addr().map(|it| it == ip).unwrap_or_default()
+                                    }),
+                                    None => false,
+                                },
+                            })
+                            .map(|(n, _)| n)
+                            .collect::<Vec<_>>();
+
+                        if !hostnames.is_empty() {
+                            hostnames.sort();
```

Comment:
Why is sorting a requirement here?

Reply:
This is mainly for unit testing.

![image](https://github.com/user-attachments/assets/ac641675-ba68-40f3-a154-64d077a2dc97)

Reply:
In that case the sort should move into the test instead of keeping it in the library.